frontend/tests/acceptance/gitlab-test.js

```diff
@@ -0,0 +1,220 @@
+import { module, test } from 'qunit'
+import { setupApplicationTest } from 'ember-qunit'
+import {
+  authenticateSession,
+  invalidateSession
+} from 'customer-center/tests/helpers/ember-simple-auth'
+import startApp from '../helpers/start-app'
+import destroyApp from '../helpers/destroy-app'
+import { selectChoose, clearSelected } from 'ember-power-select/test-support'
+import { clickTrigger } from 'ember-basic-dropdown/test-support/helpers'
+import { calendarSelect } from 'ember-power-calendar/test-support'
+import moment from 'moment'
+
+module('Acceptance | GitLab', function(hooks) {
+  setupApplicationTest(hooks)
+
+  hooks.beforeEach(function() {
+    this.application = null
+  })
+
+  hooks.beforeEach(async function() {
+    this.application = startApp()
+    let user = server.create('user', 'customer')
+    await authenticateSession(this.application, { data: user.id })
+  })
+
+  hooks.afterEach(async function() {
+    await invalidateSession(this.application)
+    destroyApp(this.application)
+  })
+
+  test('[T08] All projects and namespaces are rendered', async function(assert) {
+    await visit('/projects')
+
+    assert.equal(find('[data-test-group]').length, 3)
+    assert.equal(find('[data-test-project]').length, 12)
+
+    assert.dom('[data-test-group-name="1"]').hasText('test1')
+    assert.dom('[data-test-group-name="2"]').hasText('test2')
+    assert.dom('[data-test-group-name="3"]').hasText('test3')
+
+    assert
+      .dom('[data-test-group="0"] [data-test-project-name="0"]')
+      .hasText('project0')
+    assert
+      .dom('[data-test-group="0"] [data-test-project-name="1"]')
+      .hasText('project1')
+
+    assert
+      .dom('[data-test-group="1"] [data-test-project-name="0"]')
+      .hasText('project0')
+    assert
+      .dom('[data-test-group="1"] [data-test-project-name="1"]')
+      .hasText('project1')
+    assert
+      .dom('[data-test-group="1"] [data-test-project-name="2"]')
+      .hasText('project2')
+    assert
+      .dom('[data-test-group="1"] [data-test-project-name="3"]')
+      .hasText('project3')
+
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="0"]')
+      .hasText('project0')
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="1"]')
+      .hasText('project1')
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="2"]')
+      .hasText('project2')
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="3"]')
+      .hasText('project3')
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="4"]')
+      .hasText('project4')
+    assert
+      .dom('[data-test-group="2"] [data-test-project-name="5"]')
+      .hasText('project5')
```

Comment:
Couldn't you solve this with a loop? This is not really what I'd call "readable"

Reply:
:+1: